tuned/plugins/plugin_bootloader.py

```diff
@@ -51,6 +55,88 @@ def _get_config_options(cls):
 			"cmdline": None,
 		}
 
+	@staticmethod
+	def _options_to_dict(options, omit=""):
+		d = {}
+		omit = omit.split()
+		for o in options.split():
+			if o not in omit:
+				arr = o.split('=', 1)
+				d.setdefault(arr[0], []).append(arr[1] if len(arr) > 1 else None)
+		return d
+
+	@staticmethod
+	def _dict_to_options(d):
+		return " ".join([k + "=" + v1 if v1 is not None else k for k, v in d.items() for v1 in v])
+
+	def _rpm_ostree_status(self):
+		"""
+		Returns status of rpm-ostree transactions or None if not run on rpm-ostree system
+		"""
+		(rc, out, err) = self._cmd.execute(['rpm-ostree', 'status'], return_err=True)
+		log.debug("rpm-ostree status output stdout:\n%s\nstderr:\n%s" % (out, err))
+		if rc != 0:
+			return None
+		splited = out.split()
+		if len(splited) < 2 or splited[0] != "State:":
+			log.warn("Exceptional format of rpm-ostree status result:\n%s" % out)
+			return None
+		return splited[1]
+
+	def _wait_till_idle(self):
```

Comment:
b) is the case. If you want to change kargs and there's pending transaction from other process, it would fail. I found it out after system restart there's some startup settings and without this it raise error.